types/fh-sync.d.ts

```diff
@@ -0,0 +1,314 @@
+      metricsReportConcurrency?: number;
+      /**@type {Boolean} if cache the dataset client records using redis. This can help improve performance for the syncRecords API.
+      * Can be turned on if there are no records are shared between many different dataset clients. Default is false.*/
+      useCache?: boolean;
+      /**@type {Number} the TTL (Time To Live) value for the messages on the queue. In seconds. Default to 24 hours. */
+      queueMessagesTTL?: string;
+      /**@type {String} specify the maximum retention time of an inactive datasetClient. Any inactive datasetClient that is older than this period of time will be removed.*/
+      datasetClientCleanerRetentionPeriod?: string;
+      /** @type {String} specify the frequency the datasetClient cleaner should run. Default every hour ('1h').*/
+      datasetClientCleanerCheckFrequency?: string;
+    }
+
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
+     */
+    interface PendingWorkerBackoff {
+      strategy: string;
+      max: string;
+    }
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
+     */
+    interface AckWorkerBackoff {
+      strategy: string;
+      max: string;
+    }
+
+    /**
+     * Backoff Strategy
+     * Example: {strategy: 'exp', max: 60*1000},
+     */
+    interface SyncWorkerBackoff {
+      strategy: string;
+      max: number;
+    }
+
+    type StandardCb<T> = (err: Error | string | undefined, res: T | undefined) => void;
+    type NoRespCb = (err: Error | string | undefined) => void;
+
+    /**
+     * Options used to initialize sync for specific dataset
+     */
+    interface SyncInitOptions {
+      /**
+       * Value indicating how often the dataset client should be sync with the backend. Matches the clients default 
+       * frequency. Value in seconds
+       */
+      syncFrequency?: number,
+
+      /**
+       * Value that will be used to decide if the dataset client is not active anymore.
+       */
+      clientSyncTimeout?: number,
+
+      /**
+       * Value that determines how long it should wait for the backend list operation to complete
+       */
+      backendListTimeout?: number,
+
+      /**
+      * Specify the max wait time the dataset can be scheduled to sync again after its previous schedule, in seconds.
+      */
+      maxScheduleWaitTime?: number
+    }
+
+    /**
+     * Parameters object for request and response interceptors
+     */
+    interface SyncInterceptParams {
+      query_params: any;
+      metaData: any;
+    }
+
+    /**
+     * Connect sync server to mongo and redis 
+     * 
+     * @param mongoDBConnectionUrl 
+     * @param mongoDBConnectionOption 
+     * @param redisUrl 
+     * @param cb 
+     */
+    function connect(mongoDBConnectionUrl: string, mongoDBConnectionOption: any, redisUrl: string, cb: StandardCb<void>)
```

Comment:
You're missing the return type of  this function. TS flags it if I have all strict type checks turned on.